internal/artifacts/module_arch.go

```diff
@@ -76,11 +70,32 @@ func ExecuteSoloBuild(source, target string, extensions []string, modulesNames [
 		return errors.Wrapf(err, getBuildErrMsg(buildFailedMsg, multiBuildFailedMsg, modulesNames))
 	}
 
-	selectedModulesMap := convertListToMap(modulesNames)
+	selectedModulesMap := make(map[string]bool)
+	var selectedModulesWithDependenciesMap map[string]bool
 
-	sortedModules := sortSelectedModules(allModulesSorted, selectedModulesMap)
+	for _, moduleName := range modulesNames {
+		selectedModulesMap[moduleName] = true
+	}
 
-	err = buildSelectedModules(sourceDir, target, extensions, mtaObj, sortedModules, allDependencies, wdGetter)
+	if allDependencies {
+		selectedModulesWithDependenciesMap = make(map[string]bool)
+		for module := range selectedModulesMap {
+			err = collectSelectedModulesAndDependencies(mtaObj, selectedModulesWithDependenciesMap, module)
+			if err != nil {
+				return errors.Wrapf(err, getBuildErrMsg(buildFailedMsg, multiBuildFailedMsg, modulesNames))
+			}
+		}
+	} else {
+		selectedModulesWithDependenciesMap = selectedModulesMap
+	}
+
+	if len(selectedModulesWithDependenciesMap) > len(selectedModulesMap) {
+		logs.Logger.Infof(buildWithDependenciesMsg, "'"+strings.Join(sortModules(allModulesSorted, selectedModulesWithDependenciesMap), `',' `)+"'")
```

Comment:
Call `sortModules` before the `if`, only once (instead of doing it in 3 separate places). It will be called anyway.

Reply:
done